### PR TITLE
Expose service's status channel

### LINKED ISSUE #1033: Subscribe to shutdown event
**Is your feature request related to a problem? Please describe.**

In Jaeger we have embedded the service to our other components. We need a way to subscribe to an event that is triggered after the service is fully shutdown.

**Describe the solution you'd like**

Expose a channel that sends lifecycle messages e.g.:

* service started
* shutdown detected
* shutdown finished

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <ploffay@redhat.com>

**Description:** <Describe what has changed.>

Resolves #1033

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** < Describe what testing was performed and which tests were added.>

**Documentation:** < Describe the documentation added.>

### CODE REVIEW

File: service/service.go [hunk 1]
```diff
@@ -384,6 +399,8 @@ func (app *Application) execute(ctx context.Context, factory ConfigFactory) erro
 		zap.String("GitHash", app.info.GitHash),
 		zap.Int("NumCPU", runtime.NumCPU()),
 	)
+	app.stateChannel <- Starting
+	fmt.Println("after send")
```
Comment (owais) [inline service/service.go:403]: left over print statement

**Comment by tigrannajaryan** [comment]
@pavolloffay can you clarify how this is intended to be used? If you can please show the code that will use it.

**Comment by pavolloffay** [comment]
We would like to use this in Jaeger `all-in-one` component that combines OTEL collector and Jaeger query in a single executable.

Since the OTEL service listens on multiple terminations signals: `sigterm` and `asyncErrorChannel` we need an aggregated way to notify our service that shutdown is initiated and when it's done.

```go
        // query runs in a separate goroutine. 
	queryServer, tracerCloser := startQuery(v, logger, storageFactory, svc.ReportFatalError)
	for state := range svc.GetStateChannel() {
		switch state {
		case service.ClosingDown:
			queryServer.Close()
			tracerCloser.Close()
		case service.Closed:
			return
		}
	}
```

**Comment by pavolloffay** [comment]
@tigrannajaryan I will fix the coverage once we agree on the implementation.

**Comment by tigrannajaryan** [comment]
@pavolloffay approach LGTM.

**Comment by pavolloffay** [comment]
It should be ready for review.

**Pull request merged**